src/libsyntax/ext/expand.rs

```diff
@@ -482,6 +485,7 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
                 if let Err(msg) = validate_and_set_expn_info(def_span.map(|(_, s)| s),
                                                              false, false) {
                     self.cx.span_err(path.span, &msg);
+                    self.cx.trace_macros_diag();
```

Comment:
Also this, and the most of the others in this file.
We use dummy expansions (`kind.dummy(span)`) so that we don't have to abort.

Reply:
Same here